### PR TITLE
SFX-193: Create CircleCI Workflow

### PR DESCRIPTION
**Pull request by JoeyDeol**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+  requires_build_final: &requires_build_final
+    requires:
+      - build_final
+
+version: 2.1
+
+jobs:
+  ## Initialization jobs
+  ### generate_cache_key_file
```
Comment (jmbtutor) [inline .circleci/config.yml]: Add a newline above.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+            cd "packages/@sfx/${package}"
+            yarn build
+      - run: *archive_build
+      - save_cache:
+          key: build_cache-v{{ .Environment.CI_CACHE_VERSION }}-{{ .Branch }}-{{ .Revision }}-{{ checksum "../workflow-start" }}-{{ .Environment.CIRCLE_JOB }}-{{ epoch }}
+          paths:
+            - ~/build.tar
+
+  build_final:
```
Comment (jmbtutor) [inline .circleci/config.yml]: Add a doc comment to this.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    name: Install git submodules
+    command: git submodule update --init
+  yarn_install: &yarn_install
+    name: Install
+    command: yarn
+
+  requires_build_core: &requires_build_core
+    requires:
+      - build_core
+
```
[the remark this answers is not in the document]
Comment (JoeyDeol) [reply .circleci/config.yml]: Updated.

File: .circleci/config.yml (hunk 1 above)
Comment (JoeyDeol) [reply .circleci/config.yml]: Updated.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+      - run: *archive_build
+      - save_cache:
+          key: build_cache-v{{ .Environment.CI_CACHE_VERSION }}-{{ .Branch }}-{{ .Revision }}-{{ checksum "../workflow-start" }}-all-{{ epoch }}
+          paths:
+            - ~/build.tar
+
+  ## Unit test jobs
+
+  ### unit_test: unit_test_{package}
+  ### Runs the tests for {package}.
```
[the remark this answers is not in the document]
Comment (JoeyDeol) [reply .circleci/config.yml]: Updated.

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+      - run:
+          name: Test
+          command: |
+            package="${CIRCLE_JOB#unit_test_}"
+            package="${package//_/-}"
+
+            cd "packages/@sfx/${package}"
+            yarn test
+
+  unit_test_browser:
```
[the remark this answers is not in the document]
Comment (JoeyDeol) [reply .circleci/config.yml]: Updated.

File: .circleci/config.yml (hunk 2 above)
Comment (JoeyDeol) [reply .circleci/config.yml]: Updated.

**Pull request merged**